sdk/core/azure_core_test/src/recorded.rs

```diff
@@ -22,38 +21,61 @@ static TEST_PROXY: OnceCell<Result<Arc<Proxy>>> = OnceCell::const_new();
 ///
 /// The [Test Proxy](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md) service will be started as needed.
 /// Every `#[recorded::test]` will call this automatically, but it can also be called manually by any other test e.g., those attributed with `#[tokio::test]`.
+///
+/// This function will return `Ok(None)` if the test is running in live mode and should not use the test proxy at all.
```

Comment:
?? It looks like the function returns a `Result<TestContext>`, but this comment implies that it returns a `Result<Option<TestContext>>` - is one or the other out of sync?